### ISSUE TITLE
Reinstall (to add profiling) of libraries with C dependencies no longer possible

### ISSUE THREAD

**Issue by jystic** [haskell/cabal#3128]
With `cabal-install-1.22.4.0` I am able to rebuild/reinstall a package in place in order to enable profiling, but without having to build all the `*.o` files again.

Attempting this with `cabal-install-1.22.7.0` gives an error like this for the `text` package:
```
ar: dist/dist-sandbox-5f5bc1e1/build/cbits/cbits.p_o: No such file or directory
```

A quick look through the commits between these releases shows that https://github.com/haskell/cabal/commit/eb4687233852b5a9b5544b815eed0a44a947032d looks somewhat suspicious.

I have not yet verified if this happens on master.

**Comment by lukexi** [comment]
On it! Thanks for the cc

**Comment by lukexi** [comment]
Hi @jystic — I put a patch here for testing https://github.com/lukexi/cabal/commit/ac14bdb48b20cc2f3b919dba2462b2a7603c6029. 

Tested successfully with:
```
cabal get text
cd text-1.2.2.0
cabal configure --disable-library-profiling
cabal build
cabal configure --enable-library-profiling 
cabal build
```

**Comment by 23Skidoo** [comment]
@lukexi Confirmed that your fix works, pushed it to `master` and `1.22`. Can you please also add a test case to Cabal's PackageTests suite?

**Comment by jystic** [comment]
:100: thanks for the quick response @lukexi @23Skidoo
